frontend/app/routes/$lang+/_public+/apply+/$id+/date-of-birth.tsx

```diff
@@ -136,28 +147,35 @@ export async function action({ context: { session }, params, request }: ActionFu
     return json({ errors: parsedDataResult.error.format() });
   }
 
-  await applyRouteHelpers.saveState({ params, request, session, state: { dateOfBirth: parsedDataResult.data.dateOfBirth } });
+  await applyRouteHelpers.saveState({ params, request, session, state: parsedDataResult.data });
```

Comment:
Only save what's relevant to the state. Leave the date picker month, day and year out of it.
```suggestion
  await applyRouteHelpers.saveState({ params, request, session, state: { dateOfBirth: parsedDataResult.data.dateOfBirth, allChildrenUnder18: parsedDataResult.data.allChildrenUnder18 } });
```